File: src/test/java/com/baulsupp/oksocial/completion/UrlListTest.java
package com.baulsupp.oksocial.completion;

import org.junit.Test;

import static com.google.common.collect.Lists.newArrayList;
import static org.junit.Assert.assertEquals;

public class UrlListTest {
  @Test public void testReplacements() {
    UrlList l = new UrlList(".*", newArrayList("https://a.com/{location}", "https://a.com/here"));

    assertEquals(newArrayList("https://a.com/A", "https://a.com/B", "https://a.com/{location}",
        "https://a.com/here"), l.replace("location", newArrayList("A", "B"), true).getUrls(""));
  }

  @Test public void testReplacementsEmpty() {
    UrlList l = new UrlList(".*", newArrayList("https://a.com/{location}", "https://a.com/here"));

    assertEquals(newArrayList("https://a.com/{location}", "https://a.com/here"),
        l.replace("location", newArrayList(), true).getUrls(""));
  }
}

File: src/test/kotlin/com/baulsupp/oksocial/completion/UrlListTest.kt
package com.baulsupp.oksocial.completion

import com.google.common.collect.Lists.newArrayList
import org.junit.Test
import kotlin.test.assertEquals


class UrlListTest {
  @Test
  fun testReplacements() {
    val l = UrlList(UrlList.Match.EXACT,
        newArrayList("https://a.com/{location}", "https://a.com/here"))

    assertEquals(newArrayList("https://a.com/A", "https://a.com/B", "https://a.com/{location}",
        "https://a.com/here"), l.replace("location", newArrayList("A", "B"), true).getUrls(""))
  }

  @Test
  fun testReplacementsEmpty() {
    val l = UrlList(UrlList.Match.EXACT,
        newArrayList("https://a.com/{location}", "https://a.com/here"))

    assertEquals(newArrayList("https://a.com/{location}", "https://a.com/here"),
        l.replace("location", newArrayList(), true).getUrls(""))
  }
}
